S7.Net/Types/Int.cs

```diff
@@ -10,15 +10,15 @@ public static class Int
         /// <summary>
         /// Converts a S7 Int (2 bytes) to short (Int16)
         /// </summary>
-        public static short FromByteArray(byte[] bytes)
+        public static short FromByteArray(byte[] bytes, int offset = 0)
         {
-            if (bytes.Length != 2)
+            if (bytes.Length < offset + 2)
             {
-                throw new ArgumentException("Wrong number of bytes. Bytes array must contain 2 bytes.");
+                throw new ArgumentException($"Wrong number of bytes. Bytes array must contain at least {offset+2} bytes.");
             }
             // bytes[0] -> HighByte
             // bytes[1] -> LowByte
-            return (short)((int)(bytes[1]) | ((int)(bytes[0]) << 8));
+            return (short)(bytes[offset + 1] | (bytes[offset + 0] << 8));
```

Comment:
Not a fan of this change. It make Types.Int different than all the other types. In Sally7 I use `Span` because you can get ranges without copies, in the long run that might be a better solution here as well.

Reply:
Yes I agree with your critique. The idea was to reuse the already available big endian serialization we had. But I guess for such a one-liner we can just use hand-rolled conversion as before.

Reply:
Changed back in 4a72c35

Reply:
Yes Span is nice and all but I don't see that happening here for a long time.
I also did some test with the new Memory class and spans here in S7.NetPlus, and while it might certainly save some cpu cycles, the overall execution time difference is not measurable at all since it is just network communication and very low throughput.